### PR TITLE
Add emdaer-plugin-link

### PR DESCRIPTION
**Pull request by infiniteluke**
Open to ideas on how to make this less ugly

### CODE REVIEW

File: packages/emdaer-plugin-link/src/index.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  });
+  test('generates links with a name', async () => {
+    expect(
+      await link({
+        content: 'Anchor',
+        link: 'name thing',
+        title: 'Name Title',
+      })
+    ).toBe('<a href="#name thing" title="Name Title">Anchor</a>');
+    expect(
```
Comment (infiniteluke) [inline packages/emdaer-plugin-link/src/index.test.js]: I didn't know the best way to handle this.

File: package.json [hunk 2]
```diff
@@ -72,6 +72,7 @@
     "prettier": "1.3.1"
   },
   "dependencies": {
+    "emdaer-plugin-link": "file:packages/emdaer-plugin-link",
```
Comment (flipactual) [inline package.json]: Is this in the right place?

File: packages/emdaer-plugin-link/src/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  title,
+  name,
+  fromPlugin,
+}: {
+  content?: string,
+  link?: string,
+  title?: string,
+  name?: string,
+  fromPlugin?: PluginCall,
+}): Promise<?string> {
```
Comment (flipactual) [inline packages/emdaer-plugin-link/src/index.js]: Should we open an issue to change this pattern to `ShapeOne | ShapeTwo`?

Comment (flipactual) [inline packages/emdaer-plugin-link/src/index.js]: We'd need to detect which shape we're getting before destructuring

File: packages/emdaer-plugin-link/src/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    let href = '';
+    if (link) {
+      if (isUrl.test(link) || isRepoFilePath.test(link)) {
+        href = link;
+      } else if (!name) {
+        href = `#${link}`;
+      }
+    } else if (!name) {
+      href = '#';
+    }
```
Comment (flipactual) [inline packages/emdaer-plugin-link/src/index.js]: Maybe `getHREF(link)`?

File: packages/emdaer-plugin-link/src/index.test.js (hunk 1 above)
Comment (flipactual) [inline packages/emdaer-plugin-link/src/index.test.js]: Can you explain the confusion for me?

Comment (infiniteluke) [inline packages/emdaer-plugin-link/src/index.test.js]: If `name` and `link` is provided we either need to show both attributes or let one win. I made it so `name` wins and `link` isn't shown if both are provided. Unsure if this is even necessary.

Comment (flipactual) [inline packages/emdaer-plugin-link/src/index.test.js]: Could we not just ask users to say `link: #supdawg`?

Comment (infiniteluke) [inline packages/emdaer-plugin-link/src/index.test.js]: I think we could. That would simplify things

File: package.json (hunk 2 above)
Comment (infiniteluke) [inline package.json]: No. It's not even needed. Will remove

File: packages/emdaer-plugin-link/src/index.js (hunk 3 above)
Comment (infiniteluke) [inline packages/emdaer-plugin-link/src/index.js]: Ya. We should, I was starting to feel weird making everything optional

**Pull request merged**